lib/crypto/base36.js

```diff
@@ -0,0 +1,30 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+var crypto = require('crypto')
+
+function base36(len) {
+  var out = ''
+  while (out.length < len) {
+    var rand = crypto.randomBytes(len)
+    var randLen = rand.length
+    for (var i = 0; i < randLen; i++) {
+      var b = rand[i]
+      // 252-256 skews the base36 distribution, so skip those bytes
+      if (b < 252) {
+        out.push((b % 36).toString(36))
```

Comment:
Using `new Array(len)` w/ push doesn't work correctly because the created array is automatically of length === len, the resulting code is always empty.